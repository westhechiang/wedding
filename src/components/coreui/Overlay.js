import styled from 'styled-components';
import { theme } from '../../ui';

const Overlay = styled.div`
  width: ${props => props.width || '100%'};
  height: ${props => props.height || '100%'};
  position: ${props => props.position || 'absolute'};
  left: ${props => props.left || '0'};
  top: ${props => props.top || '0'};
  background: ${props => theme.color[props.backgroundColor] || theme.color.black};
  opacity: ${props => props.opacity || 0.8};
  z-index: ${props => props.zIndex || 1};
  transition: 1s ease-in-out opacity;
`;

export default Overlay;
