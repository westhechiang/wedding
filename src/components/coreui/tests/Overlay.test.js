import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Overlay from '../../../components/coreui/Overlay';

describe('<Overlay />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<Overlay />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
