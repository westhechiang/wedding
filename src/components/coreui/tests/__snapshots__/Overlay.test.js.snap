// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> should match its empty snapshot 1`] = `
<div
  className="sc-bdVaJa fsZOZa"
/>
`;
